usb/ehci/phy-imxrt117x.c

```diff
@@ -0,0 +1,182 @@
+	phy_debug_set, phy_debug_clr, phy_debug_tog, phy_debug0_status, phy_debug1 = 28,
+	phy_debug1_set, phy_debug1_clr, phy_debug1_tog, phy_version, phy_pll_sic = 40,
+	phy_pll_sic_set, phy_pll_sic_clr, phy_pll_sic_tog, phy_vbus_detect = 48,
+	phy_vbus_detect_set, phy_vbus_detect_clr, phy_vbus_detect_tog, phy_vbus_detect_stat,
+	phy_chrg_detect = 56, phy_chrg_detect_set, phy_chrg_detect_clr, phy_chrg_detect_tog,
+	phy_chrg_detect_stat, phy_anactrl = 64, phy_anactrl_set, phy_anactrl_clr,
+	phy_anactrl_tog, phy_loopback, phy_loopback_set, phy_loopback_clr, phy_loopback_tog,
+	phy_loopback_hsfscnt, phy_loopback_hsfscnt_set, phy_loopback_hsfscnt_clr,
+	phy_loopback_hsfscnt_tog, phy_trim_override_en, phy_trim_override_en_set,
+	phy_trim_override_en_clr, phy_trim_override_en_tog };
+
+
+static const hcd_info_t imxrt_info[] = {
+#ifdef USB_EHCI_OTG1_ENABLED
+	{
+		.type = "ehci",
+		.hcdaddr = 0x40430000,
+		.phyaddr = 0x40434000,
+		.clk = pctl_lpcg_usb,
+		.irq = usb_otg1_irq,
+	},
+#endif
+#ifdef USB_EHCI_OTG2_ENABLED
+	{
+		.type = "ehci",
+		.hcdaddr = 0x4042c000,
+		.phyaddr = 0x40438000,
+		.clk = pctl_lpcg_usb,
+		.irq = usb_otg2_irq,
+	}
+#endif
+};
+
+
+static int setClock(int dev, unsigned int state)
+{
+	platformctl_t pctl;
+
+	pctl.action = pctl_set;
+	pctl.type = pctl_lpcg;
+	pctl.lpcg.dev = dev;
+	pctl.lpcg.state = (state != 0);
+
+	return platformctl(&pctl);
+}
+
+
+int hcd_getInfo(const hcd_info_t **hcinfo)
+{
+	*hcinfo = imxrt_info;
+
+	return sizeof(imxrt_info) / sizeof(hcd_info_t);
+}
+
+
+void phy_dumpRegisters(hcd_t *hcd, FILE *stream)
+{
+	fprintf(stream, "%22s: %08x", "phy_pwd", *(hcd->phybase + phy_pwd));
+	fprintf(stream, "%22s: %08x\n", "phy_tx", *(hcd->phybase + phy_tx));
+	fprintf(stream, "%22s: %08x", "phy_rx", *(hcd->phybase + phy_rx));
+	fprintf(stream, "%22s: %08x\n", "phy_ctrl", *(hcd->phybase + phy_ctrl));
+	fprintf(stream, "%22s: %08x", "phy_status", *(hcd->phybase + phy_status));
+	fprintf(stream, "%22s: %08x\n", "phy_debug", *(hcd->phybase + phy_debug));
+	fprintf(stream, "%22s: %08x", "phy_debug0_status", *(hcd->phybase + phy_debug0_status));
+	fprintf(stream, "%22s: %08x\n", "phy_debug1", *(hcd->phybase + phy_debug1));
+	fprintf(stream, "%22s: %08x", "phy_version", *(hcd->phybase + phy_version));
+	fprintf(stream, "%22s: %08x\n", "phy_pll_sic", *(hcd->phybase + phy_pll_sic));
+	fprintf(stream, "%22s: %08x", "phy_vbus_detect", *(hcd->phybase + phy_vbus_detect));
+	fprintf(stream, "%22s: %08x\n", "phy_chrg_detect", *(hcd->phybase + phy_chrg_detect));
+	fprintf(stream, "%22s: %08x", "phy_anactrl", *(hcd->phybase + phy_anactrl));
+	fprintf(stream, "%22s: %08x\n", "phy_loopback", *(hcd->phybase + phy_loopback));
+	fprintf(stream, "%22s: %08x", "phy_loopback_hsfscnt", *(hcd->phybase + phy_loopback_hsfscnt));
+	fprintf(stream, "%22s: %08x\n", "phy_trim_override_en", *(hcd->phybase + phy_trim_override_en));
+}
+
+
+static void phy_start(hcd_t *hcd)
+{
+	uint32_t tmp;
+
+	*(hcd->base + 0x50) |= (1 << 1);
```

Comment:
This line begs for a comment - what does it do, what is register 0x50?

Reply:
I decided to remove this line, it was a controller reset (not PHY). Now it is done in `ehci.c`, so it is not necessary here, as it is not part of PHY.